mavisp/modules.py

```diff
@@ -1819,3 +1821,95 @@ def ingest(self, mutations):
             raise MAVISpMultipleError(warning=warnings,
                                       critical=[])
 
+class EnsembleAllosigmaPSN(MavispModule):
+
+    module_dir = "long_range_psn"
+    name = "long_range_psn"
+    method_dir = "allosigma2"
+    exp_files = {'allosigma_mut.txt': {'df_name': 'simple_data',
+                    'columns': ['wt_residue', 'position', 'mutated_residue', 'allosigma-mode']},
+            'results_summary.txt': {'df_name': 'ensemble_data',
+                    'columns': ['Variant_Site', 'Total_Paths']}}
+
+    def _generate_allosigma_psn_classification(self, row):
+        res_num = row['res_num']
+        allosigma_mode = row['allosigma-mode']
+        variant_sites = self.ensemble_data[self.ensemble_data['Variant_Site'].astype(str) == res_num]
+        
+       # For mutations not predicted to have effects
+        if variant_sites.empty: 
+            return 'neutral' if allosigma_mode in ['UP', 'DOWN'] else 'uncertain'
+                
+        # For mutations with predicted effects
+        total_paths = variant_sites['Total_Paths'].max()
+        if total_paths >= 1:
+            return 'damaging' if allosigma_mode in ['UP', 'DOWN'] else 'neutral'
+        else:
+            return 'uncertain'
```

Comment:
the way this is laid out looks a bit convoluted to me. For what we decided if a mutation is not classified as UP/DOWN it should always be `uncertain`. I'm not sure about line 1846 in this sense - I don't understand why it is used to switch between `damaging` and `neutral`.

Maybe a code path as follows would be simpler:

```
# unclassifiable mutation
if not allosigma_mode in ['UP', 'DOWN']:
    return 'uncertain'

# no response sites detected - PSN outcome is irrelevant
if variant_sites.empty:
    return 'neutral'

total_paths = variant_sites['Total_Paths'].max()
if total_paths >= 1:
    return 'damaging' # PSN paths found
else:
    return 'uncertain' # no PSN paths found - couldn't validate allosigma analysis
```